src/mysql_k8s_helpers.py

```diff
@@ -256,108 +253,75 @@ def create_custom_config_file(self, report_host: str) -> None:
-                if there was an issue removing not online instances
+            MySQLForceRemoveUnitFromClusterError - if there was an issue force
+                removing the unit from the cluster
         """
-        try:
-            cluster_status = self.get_cluster_status()
-            if not cluster_status:
-                raise MySQLRemoveInstancesNotOnlineRetryError("Unable to retrieve cluster status")
+        cluster_status = self.get_cluster_status()
+        if not cluster_status:
+            raise MySQLForceRemoveUnitFromClusterError()
+
+        remove_instance_options = {
+            "force": "true",
+        }
+        remove_instance_commands = (
+            f"shell.connect('{self.cluster_admin_user}:{self.cluster_admin_password}@{self.instance_address}')",
+            f"cluster = dba.get_cluster('{self.cluster_name}')",
+            f"cluster.remove_instance('{unit_address}', {json.dumps(remove_instance_options)})",
+        )
 
-            # If the cluster has no quorum, force quorum using partition of
-            # the first online instance
+        try:
             if cluster_status["defaultreplicaset"]["status"] == "no_quorum":
-                logger.warning("Cluster has no quorum")
+                logger.warning("Cluster has no quorum. Forcing quorum using this instance.")
 
-                online_member_address = [
-                    member["address"]
-                    for _, member in cluster_status["defaultreplicaset"]["topology"].items()
-                    if member["status"] == "online"
-                ][0]
-
-                logger.info(f"Forcing quorum using {online_member_address}")
                 force_quorum_commands = (
                     f"shell.connect('{self.cluster_admin_user}:{self.cluster_admin_password}@{self.instance_address}')",
                     f"cluster = dba.get_cluster('{self.cluster_name}')",
-                    f"cluster.force_quorum_using_partition_of('{self.cluster_admin_user}@{online_member_address}', '{self.cluster_admin_password}')",
+                    f"cluster.force_quorum_using_partition_of('{self.cluster_admin_user}@{self.instance_address}', '{self.cluster_admin_password}')",
                 )
 
                 self._run_mysqlsh_script("\n".join(force_quorum_commands))
 
-            self._remove_unreachable_instances()
-            self._wait_till_all_members_are_online()
-        except MySQLClientError as e:
-            # In case of an error (cluster still not stable), raise an error and retry
-            logger.warning(
-                f"Failed to remove unreachable instances on {self.instance_address} with error {e.message}"
-            )
-            raise MySQLRemoveInstancesNotOnlineRetryError(e.message)
-        except RetryError as e:
+            self._run_mysqlsh_script("\n".join(remove_instance_commands))
+        except (
+            MySQLClientError,
+            MySQLWaitUntilUnitRemovedFromClusterError,
```

Comment:
This exception and its definition can also be removed if the method that raises it isn't being called (the one Paulo asked about).

Reply:
Good catch! I just missed the call to the function that may throw this error. Added the function call in 0a939bd6332579e0fa0af593567f1b67265b4e8b